packages/grafana-ui/src/components/PageLayout/PageToolbar.tsx

```diff
@@ -45,7 +45,7 @@ export const PageToolbar: FC<Props> = React.memo(
     'aria-label': ariaLabel,
     buttonOverflowAlignment = 'right',
   }) => {
-    const styles = useStyles2(getStyles);
+    const styles = useStyles2(getStyles(!!pageIcon));
```

Comment:
don't pass props to getStyles, this destroys the memoization of styles generation.  instead return a class "noPageIcon" from the getStyles function and use that when there is no page icon

Reply:
example: https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/ToolbarButton/ToolbarButton.tsx#L74